jwst/ramp_fitting/gls_fit.py

```diff
@@ -692,52 +681,44 @@ def gls_fit(ramp_data, input_var_data,
 
     # Divide by sqrt(2) to convert the readnoise from CDS to single readout.
     rn3d = readnoise.reshape((nz, 1, 1)) * SINGLE_READOUT_RN_FACTOR
-    cov += (I * (rn3d**2 / M))
+    ramp_cov += (I * (rn3d**2 / M))
 
     # prev_slope_data must be non-negative.
     flags = prev_slope_data < 0.
     prev_slope_data[flags] = 1.
 
-    if use_extra_terms:
-        # Include two dummy axes to allow broadcasting with cov.
-        slope3d = prev_slope_data.reshape((nz, 1, 1))
-        # diagonal:
-        if gain is not None:
-            g3d = gain.reshape((nz, 1, 1))
-        else:
-            g3d = 1.
-        delta_diag = I * (slope3d * frame_time *
-                          (M - 1.) * (M - 2.) / (3. * M) +
-                          (g3d * M)**2 / 12.)
-        cov += delta_diag
-        del delta_diag
-
-    # This is the solution:  (xT @ weight @ x)^-1 @ [xT @ weight @ y]
-    # where @ means matrix multiplication.
+    # The resulting fit parameters are
+    #  (xT @ ramp_cov^-1 @ x)^-1 * [xT @ ramp_cov^-1 @ y]
+    #  = [y-intercept, slope, cr_amplitude_1, cr_amplitude_2, ...]
+    # where * means matrix multiplication.
```

Comment:
This is OK, but it was OK before, too.  That is, @ does actually mean matrix multiplication in numpy.

Reply:
You are very correct.  I had put that in before I read that @ is matrix multiplication in numpy.  Thought I'd take all of my "fixes" in this respect out, but clearly missed this one.  I've changed it back to @.